docs/api-docs/storefront/graphql/graphql-auth.md

```diff
@@ -0,0 +1,155 @@
+- [Section4](#section4)
+- [Section5](#section5)
+- [Section6](#section6)
+- [Resources](#resources)
+
+</div>
+
+Introduction
+
+## Overview
+
+GraphQL Storefront API requests are authenticated with tokens sent in the HTTP `Authorization` header.
+
+```http
+GET https://{{STORE_DOMAIN}}.com/graphql
+Authorization: Bearer {{TOKEN}}
+```
+
+## Getting tokens via API
+
+Create JWT tokens for authenticating cross-origin requests by sending a `POST` request to [`/v3/storefront/api-token`](https://developer.bigcommerce.com/api-reference/storefront/graphql-api-tokens/api-token/createtoken).
+
+```http
+POST https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/endpoint
+X-Auth-Token: {{ACCESS_TOKEN}}
+X-Auth-Client: {{CLIENT_ID}}
+Content-Type: application/json
+Accept: application/json
+
+{
+  "channel_id": 1,
+  "expires_at": 1602288000,
+  "allowed_cors_origins": [
+    "https://example.com"
+  ]
+}
+```
+
+[![Open in Request Runner](https://storage.googleapis.com/bigcommerce-production-dev-center/images/Open-Request-Runner.svg)](https://developer.bigcommerce.com/api-reference/storefront/graphql-api-tokens/api-token/createtoken#requestrunner)
+
+**Response:**
+
+```json
+{
+  "token":"...eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiJ9...",
+  "meta": {}
+}
+```
+
+## Getting tokens via handlebars
+
+Client code in BigCommerce Stencil themes can be passed a token at render time with the `{{settings.storefront_api.token}}` Handlebars object:
+
+```html
+<script>
+fetch('/graphql', {
+       method: 'POST',
+       headers: {
+           'Content-Type': 'application/json',
+           'Authorization': 'Bearer {{settings.storefront_api.token}}'
+       },
+       body: JSON.stringify({
+           query: `
+            query MyFirstQuery {...}
+  `
+// ...
+</script>
+```
+
+<div class="HubBlock--callout">
+<div class="CalloutBlock--warning">
+<div class="HubBlock-content">
+<!-- theme: warning -->
+
+### Note
+> * `1` can be passed in for the `channel_id` for generating tokens for use on the default Stencil storefront.
+> * To create a channel for a remote site, see [Create Channel].(https://developer.bigcommerce.com/api-reference/cart-checkout/channels-listings-api/channels/createchannel) in the API Reference.
+> * `allowed_cors_origins` array accepts only a single origin currently -- one token must be generated for each origin.
+> * `/storefront/api-token` endpoint requires the `Manage` `Storefront API Tokens` OAuth Scope.
+> * `storefront/api-token-customer-impersonation` endpoint requires the `Manage` `Storefront API Customer Impersonation Tokens` OAuth Scope.
+
+</div>
+</div>
+</div>
+
+## Getting customer impersonation tokens
+
+It's also possible to generate tokens for use in server-to-server interactions with a trusted consumer by POSTing to the [API Token Customer Impersonation Endpoint](https://developer.bigcommerce.com/api-reference/storefront/graphql-api-tokens/customer-impersonation-token/createtokenwithcustomerimpersonation):

```

Comment:
```suggestion
It's also possible to generate tokens for use in server-to-server interactions with a trusted consumer by sending a `POST` request to the [API Token Customer Impersonation Endpoint](https://developer.bigcommerce.com/api-reference/storefront/graphql-api-tokens/customer-impersonation-token/createtokenwithcustomerimpersonation):

```
[Code in text](https://intranet.bigcommerce.com/display/SST/Code+in+Text#CodeinText-Keywords)